core/common/src/test/scala/org/typelevel/otel4s/TextMapGetterProps.scala

```diff
@@ -69,7 +70,7 @@ class TextMapGetterProps extends ScalaCheckSuite {
   mkProps[mutable.TreeMap[String, String]]
   mkProps[mutable.SeqMap[String, String]]
   mkProps[mutable.LinkedHashMap[String, String]]
-  mkProps[concurrent.TrieMap[String, String]]
+  // mkProps[collection.concurrent.TrieMap[String, String]] // doesn't compile on JS for unknown reason
```

Comment:
`TrieMap` is not currently supported on Scala.js because it's a concurrent data structure and JS is single-threaded. Scala Native will probably have the same problem. 

e can probably shim it upstream in the scala-js repo, until then you can either disable the test or use a `Platform` like Ross suggested. We use this trick in http4s test suites.

https://github.com/http4s/http4s/blob/b22f722d23eabd6fc3e5e95c69166b454ee15ca3/tests/shared/src/test/scala/org/http4s/DecodeSpec.scala#L105-L112

Reply:
Is there a way to add that Platform trait without biting off `CrossProject.Full`?  That would be a small headache for any of the PRs in flight.  At least the ones that touch core.

Reply:
Maybe it's time to microlib it? It keeps popping up in most projects.

Reply:
I don't know how many projects want to take on a binary liability for a dozen lines.  I guess if we add a platform it's binary compatible, and if one dies the living will honor its memory with a `false`?

Can we generate this from BuildInfo?  I suppose it needs to know about crossproject.

Reply:
```
java.lang.IllegalArgumentException: A constructor or static initializer must have a void result type
```

is a weird way to say "we don't support this"

Reply:
- https://github.com/scala-js/scala-js/issues/4865

Reply:
> `TrieMap` is not currently supported on Scala.js because it's a concurrent data structure and JS is single-threaded.

I would like to note that this is not a reason for scala.js not to be able to implement it—to the contrary in fact. Since scala.js does not need to worry about concurrent modification, it should be trivial to implement by just having concurrent mechanisms not do any checks or anything special